src/child.ml

```diff
@@ -204,17 +217,35 @@ let run state comms_sock fd_sock fd_sock_path =
 			List.iter (fun fd -> Unix.close fd) fds;
 
 			(* Close the end of the pipe that's only supposed to be written to by the child process. *)
-			Opt.iter Unix.close !out_childlogging;
-
-			Opt.iter
-				(fun in_fd ->
-					let key = (match state.syslog_stdout.key with None -> Filename.basename name | Some key -> key) in
-					(* Read from the child's stdout and write each one to syslog *)
-					Unixext.lines_iter
-						(fun line ->
-							Fe_debug.info "%s[%d]: %s" key result line
-						) (Unix.in_channel_of_descr in_fd)
-				) !in_childlogging;
+			Opt.iter Unix.close !out_childlogging_stdout;
+			Opt.iter Unix.close !out_childlogging_stderr;
+
+			let running = ref true
+			in
+			while !running do
+				let (in_fd, _, _) =
+					Unix.select [Opt.unbox !in_childlogging_stdout; Opt.unbox !in_childlogging_stderr] [] [] (-1.0)
+				in
+
+				List.map (fun fd ->
+
+					let log_fd = ref state.syslog_stdout.key in
```

Comment:
Would it be possible to define these values outside the map or, even better, outside the while loop? Otherwise they are re-computed in every loop and for every value in the map.